### ISSUE TITLE
Measurable doesn't mark complete

### ISSUE THREAD

**Issue by Lissidragon** [iSoron/uhabits#1104]
## Pre-submission checklist
- [ y] I am submitting a bug report, not a feature request.
- [y ] I have searched for similar issues, but did not find any matches. (There was one feature request but didn't seem exact match)

## Description
I created a measurable task to clean the office 60 mins every month. No matter how many times I put in 60 mins, the task isn't marked completed so it doesn't hide. 

## Steps to reproduce
1. Create a new measurable task.
2. Set units to mins. Set target to 60. Set frequency to month.
3. Go to Calendar or Main view to set several days to 60.
4. Return to main view if not there already.
5. Note that task does not hide upon completing 60mins within a months time.

## System information
 - Phone: Google Pixel 4
 - Phone Operating System: Android 11
 - App version: 2.0.3

## Screenshots
![Screenshot_20210917-110853](https://user-images.githubusercontent.com/11533909/133819983-022a6044-7fc0-4c6f-b148-1801eeea3d45.png)
![Screenshot_20210917-110923](https://user-images.githubusercontent.com/11533909/133819985-b7ca6abc-5611-40d1-b665-da0cc245bdcd.png)
![Screenshot_20210917-110931](https://user-images.githubusercontent.com/11533909/133819986-540141cc-137d-4d11-9c95-930c01c319a7.png)

**Comment by sgallese** [comment]
@Lissidragon if you click the filter option, do you have "Hide completed" checked? When I have this option checked, the task is hidden for me once I complete the task.
![filter-option](https://user-images.githubusercontent.com/140911/133915719-375322c5-306c-4259-a705-87ec52e7f39a.png)
![hide-completed](https://user-images.githubusercontent.com/140911/133915720-3f68713f-a45e-41d9-8a3d-2371b1c791c0.png)

**Comment by Lissidragon** [comment]
Yes, hide completed is checked. Monthly and weekly measurable won't go away
unless I do the required time that day. But it should go away if I
completed 60min that month overall or weekly should it not?

Elissa Pendragon

**Comment by Lissidragon** [comment]
This "Test" is a monthly with 60min as target. See that hide completed is on and the task is still there despite 60min being done 3 days ago. (I toggled the hide but it still didn't go away)
![Screenshot_20210919-092321](https://user-images.githubusercontent.com/11533909/133931201-aaba6c53-5a8f-44bf-a40f-4760d70c05a0.png)

**Comment by sgallese** [comment]
This is helpful, @Lissidragon, thank you for the extra context.
I'm new to this project, and I think you've raised an interesting point.

The [hide completed filter](https://github.com/iSoron/uhabits/blob/dev/uhabits-core/src/jvmMain/java/org/isoron/uhabits/core/models/HabitMatcher.kt#L29) will hide a habit if it is [completed today](https://github.com/iSoron/uhabits/blob/dev/uhabits-core/src/jvmMain/java/org/isoron/uhabits/core/models/Habit.kt#L58-L69) or [failed today](https://github.com/iSoron/uhabits/blob/dev/uhabits-core/src/jvmMain/java/org/isoron/uhabits/core/models/Habit.kt#L71-L82).

In this case, for a measurable habit, I'm not sure if the failed today logic makes sense for the filter.
For example, let's say I need to clean my office 60 minutes a day.
If I clean it 60 minutes today, the habit will be completed.
If I only clean it 20 minutes (or 0 minutes), then the [habit is failed](https://github.com/iSoron/uhabits/discussions/1049) ([PR](https://github.com/iSoron/uhabits/pull/1052/files)).
Thus, if the hide completed filter is enabled, [this matcher](https://github.com/iSoron/uhabits/blob/dev/uhabits-core/src/jvmMain/java/org/isoron/uhabits/core/models/HabitMatcher.kt#L29) will always return false and will **always** hide the habit.

To make things more interesting, the hide completed logic only works for items completed _today_.
In this case of our current issue, we have a habit that has a frequency of each month.
One might expect, as @Lissidragon does, that the habit should be hidden if they completed this habit prior to today.
In this case, maybe the correct approach is to check if the habit has been completed (or failed) within the frequency set.
For example, if the user sets the habit frequency to one week.
If the habit has been completed (or failed) within the last week, hide the habit.
Otherwise, show the habit.

**Comment by hiqua** [comment]
The problem seems to be that we handle yes/no habits differently from measurable ones. From memory:

* Yes / no: we compute whether the current goal means that it's completed for today (even if it's not been checked explicitly today)
* Interval: we just check today's value, meaning that we will only hide it if you reached the number today, independently from what you did the past few days

**Comment by pfertyk** [comment]
Hi! Is there any update on this issue? I can see a milestone attached, is there a planned release date? :)

Let me also add that this tool is really amazing and looks and feels very professional! That's why I was surprised to find out that measurable habits are never marked as completed :D For now, I just ignore the habit when I know that I'm done with it for the week, but it would be fantastic to have it marked as done in the calendar view as well ;)

**Comment by marcoebbinghaus** [comment]
Hi, I would like to contribute a fix for that. Is anybody already working on this?

**Comment by hiqua** [comment]
@marcoebbinghaus no, go ahead!

**Comment by SimonDedman** [comment]
I believe this is the same issue but just confirming: habits with a weekly target don't highlight in the widget once the target has been exceeded.
For days (like these) when the habit has been done, but the weekly target not exceeded, I feel like [this solution](https://github.com/iSoron/uhabits/discussions/615) would be a nice fix.
![Screenshot_20220427-214550548](https://user-images.githubusercontent.com/4599748/165845262-94420d01-0504-48b9-9e5d-3b1fc13998a7.jpg)
![Screenshot_20220427-214226310](https://user-images.githubusercontent.com/4599748/165845267-e7d9fa78-9159-4771-afce-8d115ca9bd2a.jpg)
![Screenshot_20220427-214220162](https://user-images.githubusercontent.com/4599748/165845269-6e6608a4-6964-4ab1-b4cf-031574e3361b.jpg)

Thanks for a brilliant app, it's been genuinely lifechanging!
